### PR TITLE
ACS-1585: storage classes e2e tests

### PR DESCRIPTION
**Pull request by rgherghelas**

### CODE REVIEW

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+{
+        private static final String DEFAULT_SC = "Default1";
+        private static final String DEFAULT_VALUE = "default";
+        private static final String TEST_NAMESPACE = "TestNameSpace";
+        private TransactionService transactionService;
+        private AuthenticationComponent authenticationComponent;
+
+        private NodeService nodeService;
+        private NodeRef rootNode;
+        @Spy ContentStore mockContentStore;
```
Comment (tzclucian) [inline repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java:74]: The annotation should above the actual field.

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                transactionService = (TransactionService) applicationContext.getBean("TransactionService");
+                this.authenticationComponent = (AuthenticationComponent) applicationContext.getBean("authenticationComponent");
+                this.contentService = (ContentService) this.applicationContext.getBean("contentService");
+                this.contentStore = (ContentStore) ReflectionTestUtils.getField(contentService, "store");
+
+                FileContentStore fileContentStore = new FileContentStore(applicationContext, this.contentStore.getRootLocation());
+                mockContentStore = spy(fileContentStore);
+
+                when(mockContentStore.getSupportedStorageClasses()).thenReturn(Set.of(DEFAULT_SC, "Azure", "S3"));
+                ReflectionTestUtils.setField(contentService, "store", mockContentStore);
```
Comment (tzclucian) [inline repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java:91]: Why do we set in `Before` of each method a mock content store but we start most of the methods setting an actual content store?

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+                {
+                        storeRef = nodeService.createStore(storeRef.getProtocol(), storeRef.getIdentifier());
+                }
+                rootNode = nodeService.getRootNode(storeRef);
+        }
+
+        @After
+        public void afterTestMethod()
+        {
+                ReflectionTestUtils.setField(contentService, "store", contentStore);
```
Comment (tzclucian) [inline repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java:106]: Again, why do we set an actual `contentStore`  `After` each method?

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+                assertTrue("Expected DEFAULT_SC ", contentService.getStorageClassesTransitions().isEmpty());
+        }
+
+        @Test
+        public void getStorageClassesTransitions()
+        {
+                Set<String> key1 = Set.of("Default");
+                Set<String> key2 = Set.of("Warm");
```
Comment (tzclucian) [inline repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java:134]: I think you were referring to [WORM](https://aws.amazon.com/blogs/storage/tag/worm/)?

**Review by tzclucian — CHANGES_REQUESTED** [review]
```
Acceptance Criteria

1. Test end to end the storage classes support without the need of using the connectors for actual implementations of the storage classes
2. At least one content store that has storage classes implementation is tested
3. Testing from Service layer and rest api layer
4. Add / Update TAS tests for the new and updated /storage-classes Rest-API endpoint 
```

1. While using mocks we can somehow mimic a `ContentStore` I think the acceptance criteria refers to actually implementing a `ContentStore` for testing purposes similar to `RandomRoutingContentStore` from `RoutingContentStoreTest` or like `TestCenteraLikeContentStore`.
2. The tests you've added actually duplicate most of the `ContentService` tests and `FileContentStoreTests`. I think you should refer back to the point above.
3. There are no tests from the REST API layer.
4. There are no TAS tests.

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java (hunk 3 above)
Comment (rgherghelas) [reply repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java:106]: We need to reset to the default content store after each test.

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java (hunk 2 above)
Comment (aforascu) [reply repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java:92]: Moved all default checks in one test where we need to set the actual content store to test the default behavior.

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java (hunk 1 above)
Comment (aforascu) [reply repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java]: Fixed

File: repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java (hunk 4 above)
Comment (aforascu) [reply repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java]: Fixed

**Review by aforascu — APPROVED** [review]
LTGM

**Pull request merged**